exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/ProfileResources.java

```diff
@@ -220,7 +225,21 @@ public QProfiles getProfilesJSON() {
 
       final List<ProfileInfo> finishedQueries = Lists.newArrayList();
 
-      final Iterator<Map.Entry<String, QueryProfile>> range = completed.getRange(0, MAX_PROFILES);
+      //Defining #Profiles to load
+      int maxProfilesToLoad;
+      if (MaxProfiles == null) {
+        MaxProfiles = work.getContext().getConfig().getInt(ExecConstants.HTTP_MAX_PROFILES);
+      }
+
+      String maxProfilesParams = uriInfo.getQueryParameters().getFirst(MAX_QPROFILES_PARAM);
```

Comment:
Good point and I was originally doing that. But considering that this is probably going to support a whole lot more parameters in the future, I thought I'll leave a generic Context annotation, and let the implementation within the method handle that.